File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/OptionalField.js
/**
 * Data structure used to provide optional alphanumeric data for transactions.
 */

function OptionalField  () {

    this.OptionalKeyNo = 0;
    this.OptionalValue = "";

  /**
   * Sets the key number (1 to 10) for optional field.
   * @param optionalKeyNo Number
   * @return Reference to this object for method chaining. 
   */
    this.optionalKeyNo = function (optionalKeyNo) {
        this.OptionalKeyNo = optionalKeyNo;
        return this;
    }

   /**
   * Key number (1 to 10) for optional field.
   * @return Number
  **/
    this.getOptionalKeyNo = function () {
        return this.OptionalKeyNo;
    }

  /** Reference to this object for method chaining.
   * Sets the key number (1 to 10) for optional field.
   * @param optionalKeyNo Number 
   */
    this.setOptionalKeyNo = function (optionalKeyNo) {
        this.OptionalKeyNo = optionalKeyNo;
    }

  /**
   * Sets the optional field value.
   * @param optionalValue String
   * @return Reference to this object for method chaining. 
   */
    this.optionalValue = function (optionalValue) {
        this.OptionalValue = optionalValue;
        return this;
    }

   /**
   * Optional field value.
   * @return String
  **/
    this.getOptionalValue = function () {
        return this.OptionalValue;
    }

  /** Reference to this object for method chaining.
   * Sets the optional field value.
   * @param optionalValue String 
   */
    this.setOptionalValue = function (optionalValue) {
        this.OptionalValue = optionalValue;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/OptionalField.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Data structure used to provide optional alphanumeric data for transactions.
 */
public class OptionalField   {
  @SerializedName("OptionalKeyNo")
  private Integer optionalKeyNo = null;

  @SerializedName("OptionalValue")
  private String optionalValue = null;

  /**
   * Sets the key number (1 to 10) for optional field.
   * @param optionalKeyNo
   * @return 
   */
  public OptionalField optionalKeyNo(Integer optionalKeyNo) {
    this.optionalKeyNo = optionalKeyNo;
    return this;
  }

   /**
   * Key number (1 to 10) for optional field.
   * @return optionalKeyNo
  **/
  public Integer getOptionalKeyNo() {
    return optionalKeyNo;
  }

  /**
   * Sets the key number (1 to 10) for optional field.
   * @param optionalKeyNo 
   */
  public void setOptionalKeyNo(Integer optionalKeyNo) {
    this.optionalKeyNo = optionalKeyNo;
  }

  /**
   * Sets the optional field value.
   * @param optionalValue
   * @return 
   */
  public OptionalField optionalValue(String optionalValue) {
    this.optionalValue = optionalValue;
    return this;
  }

   /**
   * Optional field value.
   * @return optionalValue
  **/
  public String getOptionalValue() {
    return optionalValue;
  }

  /**
   * Sets the optional field value.
   * @param optionalValue 
   */
  public void setOptionalValue(String optionalValue) {
    this.optionalValue = optionalValue;
  }
}

